### PR TITLE
Feature/sprint1/story3 host create lobby

### PR DESCRIPTION
**Pull request by dlhxu**
### Frontend changes
- Added lobby components, with functionality to determine if a user is a host or a guest and run different hooks, render different components based on that
- Added new socket io code to create a room and request a room ID
- Added routing for lobby in app.js
### Backend changes
- Added socket io code to send a generated room ID back to the host socket that requested it
- Added stubbed implementation of a room ID generator function to be continued in future sprints
- Added a utils.js file 
### testing
- manual testing done to confirm that the correct buttons appear under the correct flows (host vs guest)
- manual testing done to confirm proper integration with the features of Story 1 & Story 2

### CODE REVIEW

File: wewatch-client/src/components/Lobby/Lobby.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    const [hostName, setHostName] = useState('');
+    const [roomId, setRoomId] = useState('');
+    const [users, setUsers] = useState([]);
+
+    useEffect(() => {
+        socket = io(ENDPOINT);
+
+        const { name, room } = queryString.parse(location.search);
+        
+        if (room === undefined){ // host user does not have room ID in query params
```
Comment (jeffreyquiambao) [inline wewatch-client/src/components/Lobby/Lobby.js:23]: nitpick but move the comment above the if?

File: wewatch-client/src/components/Lobby/Lobby.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        socket.on("roomData", ({ users }) => {
+            setUsers(users);
+        });
+    }, []);
+
+    return (
+        <div className="outerContainer">
+            <div className="container">
+                <h1>Welcome to {hostName}'s Lobby</h1>
+                <h2>lobby ID: {roomId} </h2>
```
Comment (jeffreyquiambao) [inline wewatch-client/src/components/Lobby/Lobby.js:62]: use "Lobby ID:" for caps consistency

**Review by jeffreyquiambao — CHANGES_REQUESTED** [review]

**Review by jeffreyquiambao — APPROVED** [review]

File: server/index.js [hunk 3]
```diff
@@ -21,6 +21,20 @@ io.on('connection', (socket) => {
         }
     });
 
+    // TODO add a way for host identifier to be stored and broadcasted to room?
```
Comment (billxsheng) [inline server/index.js:24]: is this TODO done?

File: server/utils.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const generateRoomId = () => {
```
Comment (rosakng) [inline server/utils.js:1]: 👍

File: wewatch-client/src/App.js [hunk 5]
```diff
@@ -1,16 +1,21 @@
 import React from 'react';
-import { BrowserRouter as Router, Route } from "react-router-dom";
 
-import Chat from 'components/Chat/Chat';
-import Join from 'components/Join/Join';
+import Chat from './components/Chat/Chat';
```
Comment (rosakng) [inline wewatch-client/src/App.js:3]: does this work without the './' ? We should have a file called 
jsconfig.json that that has a setting for importing with src being the highest level. so you wouldnt have to make these changes

**Review by varoongupta — APPROVED** [review]

File: wewatch-client/src/App.js (hunk 5 above)
Comment (dlhxu) [reply wewatch-client/src/App.js:3]: Let me give that a try

File: server/index.js (hunk 3 above)
Comment (dlhxu) [reply server/index.js:24]: Completed and pushed in new commits

**Review by rosakng — APPROVED** [review]
🚀 LGTM

**Review by billxsheng — APPROVED** [review]

**Pull request merged**